Gopkg.toml

```diff
@@ -50,32 +50,40 @@ required = [
   name = "github.com/giantswarm/micrologger"
 
 [[constraint]]
-  branch = "master"
+  branch = "k8s-1.15"
   name = "github.com/giantswarm/operatorkit"
 
 [[constraint]]
-  branch = "master"
+  branch = "k8s-1.15"
   name = "github.com/giantswarm/tenantcluster"
 
+[[override]]
+  version = "v1.5.2"
+  name = "github.com/russross/blackfriday"
+
 [[constraint]]
   name = "k8s.io/api"
-  version = "kubernetes-1.13.2"
+  version = "kubernetes-1.15.0"
 
 [[constraint]]
   name = "k8s.io/apiserver"
-  version = "kubernetes-1.13.2"
+  version = "kubernetes-1.15.0"
 
 [[constraint]]
   name = "k8s.io/apiextensions-apiserver"
-  version = "kubernetes-1.13.2"
+  version = "kubernetes-1.15.0"
 
 [[constraint]]
   name = "k8s.io/apimachinery"
-  version = "kubernetes-1.13.2"
+  version = "kubernetes-1.15.0"
 
 [[constraint]]
   name = "k8s.io/client-go"
-  version = "kubernetes-1.13.2"
+  version = "kubernetes-1.15.0"
+
+[[override]]
+  name = "k8s.io/helm"
+  version = "v2.13.0"
```

Comment:
@tomahawk28 @rossf7 @xh3b4sd will this work with 2.12 tiller? or should it stay same as tiller and updated in same go as tiller? or do we plan to rollout these operator deps upgrades only once tiller is upgraded?

Reply:
That is not our problem right now. This was the only version I could make `dep` happy somehow and gather all dependencies. For now we should focus on updating the k8s deps. Then people can go with their desired next step. At this point `dep` is pure magic.

Reply:
@sslavic Good catch. I think this can cause problems. The Helm client and server are tightly coupled and can error if the major or minor version is different.

IMO would be better to stay on `2.12.0` and then upgrade to `2.14.1` (latest helm) as a separate task. If not we'll need to do the helm upgrade as part of this upgrade.

Reply:
Do you say it is absolutely necessary to stay with `helm@2.12.0`? This is new to me. And I think we cannot even go to `k8s@1.15` with this constraint.

Reply:
> This was the only version I could make dep happy somehow and gather all dependencies.

@xh3b4sd In that case, we need to use helm `2.13.0` everywhere and increment the version here. 
https://github.com/giantswarm/helmclient/blob/194739f9cc7e4c910fd66955001469ca9d516201/spec.go#L18

Reply:
> Do you say it is absolutely necessary to stay with `helm@2.12.0`? This is new to me. And I think we cannot even go to `k8s@1.15` with this constraint.

Not sure is it Helm CLI or Tiller but one of them is enforcing that they have same major.minor version. E.g. try using 2.14.x Helm CLI now just to `helm list` releases on any of the current clusters which have 2.12.x Tiller installed.

This check could be part of Helm CLI `cmd` package logic, and not of "library" packages that we use. I doubt that the check is there for no reason. Knowing this, using 2.13.x Helm library against 2.12 Tiller might work, if we're lucky - no pre-flight check, also we may be lucky not to be affected of any incompatibilities; still luck is not something I'd rely on.

Reply:
AIUI the tight coupling is because they use proto buffers and the specs are "loosely versioned'. There are checks for version compatibility.

Like @sslavic says we might get lucky using `2.13.0` but we'd need to manually test. Yes, this is a pain and its yet another reason why we want to remove tiller from our stack.

@xh3b4sd Let me know if you want to do a call and discuss.